crates/testenv/src/lib.rs

```diff
@@ -168,22 +168,36 @@ impl TestEnv {
     }
 
     /// This method waits for the Electrum notification indicating that a new block has been mined.
-    pub fn wait_until_electrum_sees_block(&self) -> anyhow::Result<()> {
+    /// `timeout` is the maximum amount of seconds we want to wait for a response from Electrsd.
+    pub fn wait_until_electrum_sees_block(&self, timeout: u32) -> anyhow::Result<()> {
```

Comment:
Can we please make `timeout` a `Duration`?

You can instantiate an `Instant` before the loop and check `Instant::elapsed` within the loop to determine if we need to break.

https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed